src/bridge_stream.rs

```diff
@@ -0,0 +1,312 @@
+
+    fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        let this = self.project();
+        debug!("Interfaces: {:?}", this.interfaces.len());
+        let interfaces: &mut VecDeque<BridgedInterface> = this.interfaces;
+        let config: &mut Config = this.config; //TODO use the Self {} extractor
+
+        let mut gather_to: Option<SystemTime> = None;
+        for iface in interfaces.iter_mut() {
+            let mut was_delayed = false;
+
+            if iface.complete {
+                return Poll::Ready(None);
+            }
+            if let Some(mut existing_delay) = iface.delaying.take() { //Check the interface for a delay..
+                if let Poll::Pending = Pin::new(&mut existing_delay).poll(cx) { //still delayed?
+                    debug!("Delaying");
+                    iface.delaying = Some(existing_delay);
+                    continue; // do another iteration on another iface
+                }
+                was_delayed = true;
+            }
+            let mut existing_future = iface.pending.take().unwrap_or_else(|| PacketFuture::new(config, &iface.handle));
+            match Pin::new(&mut existing_future).poll(cx) {
+                Poll::Pending => {
+                    debug!("Pending");
+                    iface.pending = Some(existing_future);
+                    continue;
+                    //return Poll::Pending;
+                }
+                Poll::Ready(Err(e)) => {
+                    return Poll::Ready(Some(Err(e)));
+                }
+                Poll::Ready(Ok(None)) => {
+                    debug!("Interface has completed");
+                    iface.complete = true;
+                    continue;
+//                    let res = gather_packets(interfaces, gather_to);
+//                    if res.is_empty() {
+//                        return Poll::Ready(None);
+//                    } else {
+//                        return Poll::Ready(Some(Ok(res)));
+//                    }
+                }
+                Poll::Ready(Ok(Some(v))) => {
+                    if v.is_empty() && !was_delayed {
+                        iface.delaying = Some(tokio_timer::delay_for(*config.retry_after()));
+                        continue;
+                    }
+                    if let Some(p) = v.last() {
+                        gather_to = gather_to.map(|ts| {
+                            std::cmp::min(ts, *p.timestamp())
+                        }).or(Some(*p.timestamp()));
+                    }
+                    trace!("Adding {} packets to current", v.len());
+                    iface.current.extend(v);
+                }
+            }
+        }
+
+        interfaces.retain(|iface| { //drop and complete interfaces
+            if iface.complete {
+                return false;
+            }
+            return true;
+        });
+
+        if interfaces.is_empty() {
+            let res = gather_packets(interfaces, gather_to);
```

Comment:
if interfaces is empty, nothing will be gathered

Reply:
derp